genrl/agents/deep/a2c/a2c.py

```diff
@@ -66,7 +66,24 @@ def _create_model(self) -> None:
         state_dim, action_dim, discrete, action_lim = get_env_properties(
             self.env, self.network
         )
-        if isinstance(self.network, str):
+        if isinstance(self.network, str) and self.shared_layers is not None:
+            self.ac = get_model("ac", self.network + "s")(
+                state_dim,
+                action_dim,
+                shared_layers=self.shared_layers,
+                policy_layers=self.policy_layers,
+                value_layers=self.value_layers,
+                val_type="V",
+                discrete=discrete,
+                action_lim=action_lim,
+            ).to(self.device)
+            actor_params = list(self.ac.shared.parameters()) + list(
+                self.ac.actor.parameters()
+            )
```

Comment:
This idea is nice but it would be better if you could perhaps shift this to the `SharedActorCritic` Class itself.

Reply:
> This idea is nice but it would be better if you could perhaps shift this to the `SharedActorCritic` Class itself.

Added another function in all of the ACs for this, hence it becomes a part of the AC class. Sounds good?